src/weather-bar.ts

```diff
@@ -89,6 +92,8 @@ export class WeatherBar extends LitElement {
       const showPrecipitationAmounts = this.show_precipitation_amounts && !skipLabel;
       const showPrecipitationProbability = this.show_precipitation_probability && !skipLabel;
       const { hour, date, temperature } = this.temperatures[i];
+      const numericTemperature = parseFloat(temperature);
+      const roundedTemperature = this.round_temperatures ? Math.round(numericTemperature) : temperature;
```

Comment:
Nitpick: let's rename this to `displayTemperature` since it's not necessarily rounded.

```suggestion
      const displayTemperature = this.round_temperatures ? Math.round(numericTemperature) : temperature;
```